app/services/forest_liana/permissions_checker.rb

```diff
@@ -0,0 +1,62 @@
+require 'date'
+
+module ForestLiana
+  class PermissionsChecker
+    @@permissions = nil
+    @@last_retrieve = nil
+
+    def initialize(resource, permission_name)
+      @collection_name = ForestLiana.name_for(resource)
+      @permission_name = permission_name
+
+      @expiration_in_seconds = !ENV['FOREST_PERMISSIONS_EXPIRATION_IN_SECONDS'].nil? ?
+        ENV['FOREST_PERMISSIONS_EXPIRATION_IN_SECONDS'].to_i :
+        3600
+    end
+
+    def is_allowed
+      if !@@permissions ||
+        !@@permissions[@collection_name] ||
+        !@@permissions[@collection_name]["collection"]
+        return false
+      end
+
+      return @@permissions[@collection_name]["collection"][@permission_name];
+    end
+
+    def retrieve_permissions
+      @@permissions = PermissionsGetter.new.perform
+      @@last_retrieve = DateTime.now
+    end
+
+    def date_difference_in_seconds(date1, date2)
+      ((date1 - date2) * 24 * 60 * 60).to_i
+    end
+
+    def is_permission_expired
+      current_time = DateTime.now
+
+      return true if @@last_retrieve.nil?
```

Comment:
why isn't it the first line of the function? What is the point to instantiate `current_time` in this case?